include/boost/date_time/time_duration.hpp

```diff
@@ -144,10 +144,26 @@ namespace date_time {
     {
       return duration_type(ticks_ * (-1));
     }
+    duration_type abs() const
```

Comment:
These functions seem like a nice addition.  Unfortunately no one will no about them without documentation.  The docs are in boost book xml which is a bit clunky to modify, but it's pretty obvious.  You should be able to find the accessor table for time_duration add trivially add these.